### PR TITLE
Fix #1721

### LINKED ISSUE #1721: Heroku deployment is broken
Hello! I tried launching a version of the Editor on Heroku yesterday but ran into an error with the (now defunct) mongolab addon.

mLab error

#### Nature of issue?

- Found a bug

#### Details about the bug: 

- Web browser and version: Firefox 84
- Operating System: macOS Catalina
- Steps to reproduce this bug: Click the big purple button in `developer_docs/deployment.md`.

### PR DESCRIPTION
**Pull request by nickmcintyre**
Fixes #1721

I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] is from a uniquely-named feature branch and has been rebased on top of the latest `develop` branch. (If I was asked to make more changes, I have made sure to rebase onto `develop` then too)
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`

### CODE REVIEW

File: developer_docs/deployment.md [hunk 1]
```diff
@@ -41,13 +41,17 @@ _Note_: The installation steps assume you are using a Unix-like shell. If you ar
...
-
-If you are interested in hosting and deploying your own p5.js Web Editor instance, you can! It would be the same as the official editor instance at editor.p5js.org, except with a different domain, and you would be in charge of the maintenance. We recommend using Heroku as you can host it for free.
-
-1. Sign up for a free account at: [Heroku](https://www.heroku.com/)
-2. Click here: [![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy?template=https://github.com/processing/p5.js-web-editor/tree/develop)
-3. Enter a unique *App name*, this will become part of the url (i.e. https://app-name.herokuapp.com/)
-4. Update any configuration variables, or accept the defaults for a quick evaluation (they can be changed later to enable full functionality)
-5. Click on the "Deploy app" button
-6. When complete, click on the "View app" button
+## Self Hosting - Heroku + Atlas Deployment
```
Comment (catarak) [inline developer_docs/deployment.md:44]: Were you able to test this? For the web editor in development, staging, and production, I have an extra step with using Atlas which is whitelisting certain IP addresses that can connect to the database. If it seems to be necessary here, would you mind adding it to the documentation? Thank you!

**Review by catarak — CHANGES_REQUESTED** [review]

File: developer_docs/deployment.md (hunk 1 above)
Comment (nickmcintyre) [reply developer_docs/deployment.md:44]: Yep and happy to do so. I got up and running by allowing access from anywhere, then got the [Fixie Socks](https://elements.heroku.com/addons/fixie-socks) addon configured. Currently trying to figure out why Fixie Socks isn't automatically provisioned despite adding it to `app.json`... working on it.

Comment (nickmcintyre) [reply developer_docs/deployment.md:44]: Yes and no. The setup works fine with all IP addresses allowed in Atlas (yikes) but I wasn't clever enough to limit access to a single dyno/project in Heroku. Thought I had with the [Fixie Socks addon](https://elements.heroku.com/addons/fixie-socks) but no luck. The [ObjectRocket for MongoDB addon](https://elements.heroku.com/addons/ormongo) works fine and starts at $46/mo.

What do you think about a rewrite to the effect of "You can demo your own instance of the Editor for free using Atlas, but you really should use something like ObjectRocket for production. Here's how you do both."?

Comment (catarak) [reply developer_docs/deployment.md:44]: Huh, that's surprising that Fixie Socks didn't work! It seems like this is just what it's for...

I don't want to add too many configuration suggestions to the guide—maybe it would be helpful to link to this article on [Using MongoDB Atlas with Heroku](https://developer.mongodb.com/how-to/use-atlas-on-heroku)?

Comment (nickmcintyre) [reply developer_docs/deployment.md:44]: I thought so as well. Tried again fresh this morning with both the Fixie Socks and [QuotaGuard Static IP's](https://elements.heroku.com/addons/quotaguardstatic) addons and wound up hitting walls with each. Simply copying the static IP’s provided won’t do it.

TL;DR ¯\\_(ツ)_/¯

The article you shared is straightforward enough to follow and aligns with this [nice write up from freeCodeCamp](https://www.freecodecamp.org/news/deploying-a-mern-application-using-mongodb-atlas-to-heroku/). However, both guides punt on the specifics of configuring the network addons. I thought I was in crazy town, but at least a few other people have hit similar stumbling blocks.

- [How to use Fixie Socks (Heroku add-on) to connect to mongodb via mongoose](https://stackoverflow.com/questions/61376584/how-to-use-fixie-socks-heroku-add-on-to-connect-to-mongodb-via-mongoose) (I tried and failed to follow the QuotaGuard documentation)
- [Connecting to mongodb using mongoose and Fixie (Heroku add-on)](https://stackoverflow.com/questions/42229390/connecting-to-mongodb-using-mongoose-and-fixie-heroku-add-on) (one commenter gave up on Fixie Socks)

The Editor crashed on start up when I tried the [QuotaGuard wrapper approach](https://support.quotaguard.com/support/solutions/articles/12000066411-how-to-connect-to-mongodb-using-quotaguard). I think this is because the addon isn't (or can't be?) provisioned automatically, so the proxy isn't available and MongoDB fails to connect.

I *think* using Fixie Socks requires modifying `server.js` along the lines of their [examples for other databases](https://usefixie.com/documentation/socks/). Dunno if this use case warrants adding a dependency and more code to maintain.

Comment (catarak) [reply developer_docs/deployment.md:44]: Thanks for trying this again! It's really frustrating that this is so difficult to configure. Maybe for now the best approach is to get this working with MongoDB by whitelisting all IPs, and then in the future figure out how to make a more secure solution.

Comment (nickmcintyre) [reply developer_docs/deployment.md:44]: I am a total space cadet–sorry for the delay! Just updated the guide to include the article you shared along with a note of caution. [Here](https://p5-editor.herokuapp.com/) is a working version of the Editor launched from my fork.
